File: src/py/Sample/Usage/Sample.py
##############################################################################
##
##       Copyright (C) 2017 Frank Eskesen.
##
##       This file is free content, distributed under the GNU General
##       Public License, version 3.0.
##       (See accompanying file LICENSE.GPL-3.0 or the original
##       contained within https://www.gnu.org/licenses/gpl-3.0.en.html)
##
##############################################################################
##
## Title-
##       Sample.py
##
## Purpose-
##       Sample operator Commands
##
## Last change date-
##       2017/01/01
##
##############################################################################
from __future__ import print_function

from lib.Command import command
from lib.Debug import Debug, debugf, tracef

##############################################################################
## __Command class (Sample command)
##############################################################################
class __Command:
    @staticmethod
    def run(argv):
        debugf("Sample command")
        return 0

command['sample'] = __Command

File: src/cpp/Sample/Generic/Sample.cpp
//----------------------------------------------------------------------------
//
//       Copyright (c) 2007 Frank Eskesen.
//
//       This file is free content, distributed under the MIT license.
//       (See accompanying file LICENSE.MIT or the original contained
//       within https://opensource.org/licenses/MIT)
//
//----------------------------------------------------------------------------
//
// Title-
//       Sample.cpp
//
// Purpose-
//       Instantiate Sample object methods.
//
// Last change date-
//       2007/01/01
//
//----------------------------------------------------------------------------
#include <iostream>
using namespace std;

#include "Sample.h"

//----------------------------------------------------------------------------
// Constants for parameterization
//----------------------------------------------------------------------------
#define __SOURCE__       "SAMPLE  " // Source file

//----------------------------------------------------------------------------
//
// Method-
//       Sample::~Sample
//
// Purpose-
//       Destructor.
//
//----------------------------------------------------------------------------
   Sample::~Sample( void )          // Destructor
{
   cout << "!" << endl;
}

//----------------------------------------------------------------------------
//
// Method-
//       Sample::Sample
//
// Purpose-
//       Default Constructor.
//
//----------------------------------------------------------------------------
   Sample::Sample( void )           // Default constructor
{
   cout << "Hello ";
}

//----------------------------------------------------------------------------
//
// Subroutine-
//       sample
//
// Purpose-
//       SAMPLE test
//
//----------------------------------------------------------------------------
int sample(int, char**);            // (Not very far) Forward reference
int                                 // Error count
   sample(int, char**)              // MALLOC test
//   int               argc,        // Argument count (Unused)
//   char*             argv[])      // Argument array (Unused)
{
   int                 errorCount= 0; // Error count
   Sample              object;      // Sample object

   //-------------------------------------------------------------------------
   // Drive the tests
   //-------------------------------------------------------------------------
   cout << "sample world";

   return errorCount;
}

